crates/storage/src/interface.rs

```diff
@@ -1053,6 +1053,11 @@ impl<S: StorageAdaptor> MainInterface<S> {
             }
         };
 
+        // We have to serialise here rather than send the Action itself, as the SDK
+        // has no knowledge of the Action type, and cannot use it as it would be a
+        // circular dependency.
```

Comment:
ideally, we define `calimero-storage-primitives` at this point, but this is fine too